### PR TITLE
Document size limits for app manifest fields

### PR DESCRIPTION
**Pull request by npclaudiu**
Document interfaces relevant to the app manifest to include current size limits applied to its fields when parsed on the device.

Signed-off-by: Claudiu Nedelcu <cnedelcu@fitbit.com>

### CODE REVIEW

File: src/componentManifest.ts [hunk 1]
```diff
@@ -13,34 +13,155 @@ import { apiVersions } from './sdkVersion';
 const PLUGIN_NAME = 'componentManifest';
 const manifestPath = 'manifest.json';
 
+/**
+ * Descriptor for localized resources.
+ *
+ * On device, `locale` is capped at 64 bytes,
```
Comment (Hexxeh) [inline src/componentManifest.ts]: Which field does this refer to? It looks like the three below are covered by other comments.

File: src/componentManifest.ts [hunk 2]
```diff
@@ -13,34 +13,155 @@ import { apiVersions } from './sdkVersion';
...
-  // interface describes the manifest as the device expects it,
-  // independent of data structures elsewhere. We want the build to
-  // fail if the AppType enum changes such that it becomes incompatible
-  // with what the device expects.
-  appType: 'app' | 'clockface';
+
+  /**
+   * JS App type.
+   *
+   * If none is specified, it will default to 'app' for backwards compatibility.
```
Comment (Hexxeh) [inline src/componentManifest.ts]: This is the FW behaviour?

File: src/componentManifest.ts (hunk 1 above)
Comment (npclaudiu) [reply src/componentManifest.ts]: I forgot to remove it from here after moving the comment to the actual field. Fixed, thanks.

File: src/componentManifest.ts (hunk 2 above)
Comment (npclaudiu) [reply src/componentManifest.ts]: Yes, for the current app manifest version.

**Pull request merged**